docs/docs/optimizing-performance.md

```diff
@@ -0,0 +1,118 @@
+---
+id: optimizing-performance
+title: Optimizing Performance
+permalink: docs/optimizing-performance.html
+---
+
+Internally, React uses several clever techniques to minimize the number of costly DOM operations required to update the UI. For many applications, using React will lead to a fast user interface without doing much work to specifically optimize for performance. Nevertheless, there are several ways you can speed up your React application.
+
+## Use The Production Build
+
+If you're benchmarking or experiencing performance problems in your React apps, make sure you're testing with the [minified production build](/react/downloads.html). The development build includes extra warnings that are helpful when building your apps, but it is slower due to the extra bookkeeping it does.
```

Comment:
This should explain how to get production build in different environments. "Downloads" page doesn't do this now.

* For Create React App, you need to run `npm run build` and follow the instructions.
* For single-file builds, we offer `.min.js`
* For Browserify, you need to run it with `NODE_ENV=production`, and it will envify React
* For Webpack, you need to add this to `plugins` in your production config:

  ```js
  new webpack.DefinePlugin({
    'process.env': {
      NODE_ENV: JSON.stringify('production')
    }
  }),
  new webpack.optimize.UglifyJsPlugin()
```